src/ggl-setup.py

```diff
@@ -0,0 +1,524 @@
+import argparse
+import grp
+import pwd
+import json
+import os
+import re
+import uuid
+import shutil
+import subprocess
+import platform
+import requests
+import zipfile
+from typing import Any, Dict, List, Literal, Optional, Sequence, Tuple
+from uuid import uuid1
+from boto3 import client
+from botocore.exceptions import ClientError
+import time
+import logging
+import yaml
+from subprocess import run
+from pathlib import Path
+from typing import Sequence, Optional, Any, Dict, List, Literal, Optional, Sequence, NamedTuple
+
+S3_ARTIFACT_DIR = "artifacts"
+DEVICE_PATH = "/var/lib/greengrass/device.pem.crt"
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

 The absolute path starts at the root and moves down through each file. While traversing the path, if the folder name or location is different than the file path mentioned in the code, it causes a break. Since most of the computers have different directory structures, using absolute path in your code is likely to cause issues in your computational environment. We recommend using relative paths, because it only relies on relevant part of the directory structure being the same making it reliable to work with in computational environment

Similar issue at line numbers 26 and 27.